From 11831cd50b283a59e59573eb823ecf5500280888 Mon Sep 17 00:00:00 2001
From: Rajat Khandelwal <prongs@apache.org>
Date: Sun, 17 Apr 2016 06:46:30 +0530
Subject: [PATCH 0580/1164] CDH-39611 : HIVE-13500 : Fix OOM with explan
 output being logged (Rajat Khandelwal, reviewed
 by Amareshwari)

(cherry picked from commit 68dc9ddd3e304ffc523e92d51990b15a2884ba10)

Change-Id: I153849d3b145b2296773d494e178bb4c6d24e959
---
 ql/src/java/org/apache/hadoop/hive/ql/Driver.java |   17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/Driver.java b/ql/src/java/org/apache/hadoop/hive/ql/Driver.java
index c4872be..ff900be 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/Driver.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/Driver.java
@@ -494,15 +494,16 @@ public int compile(String command, boolean resetTaskIds) {
         }
       }
 
-      if (conf.getBoolVar(ConfVars.HIVE_LOG_EXPLAIN_OUTPUT) ||
-          conf.isWebUiQueryInfoCacheEnabled()) {
+      if (conf.getBoolVar(ConfVars.HIVE_LOG_EXPLAIN_OUTPUT)) {
         String explainOutput = getExplainOutput(sem, plan, tree.dump());
-        if (conf.getBoolVar(ConfVars.HIVE_LOG_EXPLAIN_OUTPUT)) {
-          LOG.info("EXPLAIN output for queryid " + queryId + " : "
-            + explainOutput);
-        }
-        if (conf.isWebUiQueryInfoCacheEnabled()) {
-          queryDisplay.setExplainPlan(explainOutput);
+        if (explainOutput != null) {
+          if (conf.getBoolVar(ConfVars.HIVE_LOG_EXPLAIN_OUTPUT)) {
+            LOG.info("EXPLAIN output for queryid " + queryId + " : "
+              + explainOutput);
+          }
+          if (conf.isWebUiQueryInfoCacheEnabled()) {
+            queryDisplay.setExplainPlan(explainOutput);
+          }
         }
       }
       return 0;
-- 
1.7.9.5

